teacher.rb

```diff
@@ -0,0 +1,17 @@
+require_relative 'person'
+
+# Represents a teacher with additional specialization.
+class Teacher < Person
+  attr_reader :specialization
+
+  def initialize(id, age = 18, name = 'Unknown', parent_permission: true, specialization: 'Ruby on Rails')
+    # Call the parent's constructor using super
+    super(id, age, name, parent_permission)
```

Comment:
- [ ] Kindly assign the parent_permission correctly.

  ```
  super(id, age, name, parent_permission: parent_permission)
  ```

Reply:
Thanks 👍